src/boot/efi/xbootldr.c

```diff
@@ -133,7 +133,7 @@ static EFI_STATUS try_gpt(
                 EFI_PARTITION_ENTRY *entry =
                                 (EFI_PARTITION_ENTRY *) ((uint8_t *) entries + gpt.gpt_header.SizeOfPartitionEntry * i);
 
-                if (memcmp(&entry->PartitionTypeGUID, XBOOTLDR_GUID, sizeof(entry->PartitionTypeGUID)) != 0)
+                if (memcmp(&entry->PartitionTypeGUID, &MAKE_GUID(XBOOTLDR), sizeof(entry->PartitionTypeGUID)) != 0)
```

Comment:
Maybe you can fold in @poettering's https://github.com/systemd/systemd/pull/25350 to this commit, except make it lass by value, and then you can clean up all these memcmps to do the right thing too.

Then the rule is: GUIDs always pass by reference, with the one exception being EFI function pointer prototypes that we can't change.